src-unix/native/winsize.c

```diff
@@ -10,7 +11,19 @@
 CAMLprim value caml_notty_winsize (value vfd) {
 #ifdef _WIN32
   (void) vfd;
-  caml_failwith("not implemented on windows");
+  HANDLE hConsole = GetStdHandle(STD_OUTPUT_HANDLE);
+  if (hConsole == INVALID_HANDLE_VALUE) {
+    caml_failwith("Failed to get a console handle on Windows");
+  }
+
+  CONSOLE_SCREEN_BUFFER_INFO csbi;
+  if (GetConsoleScreenBufferInfo(hConsole, &csbi)) {
+    int columns = csbi.srWindow.Right - csbi.srWindow.Left + 1;
+    int rows = csbi.srWindow.Bottom - csbi.srWindow.Top + 1;
+    return Val_int ((columns << 16) + ((rows & 0x7fff) << 1));
+  } else {
+    caml_failwith("Failed to get console screen buffer info on Windows");
+  }
```

Comment:
This appears to be a standard solution, adapted from: https://stackoverflow.com/questions/6812224/getting-terminal-size-in-c-for-windows